src/plugins/diveprofile/index.js

```diff
@@ -1,66 +1,215 @@
-function diveprofile(name, deps) {
-  console.log('This is where DiveProfile plugin.');
-  //instance variables
-  this.cockpit = deps.cockpit;
-  this.global = deps.globalEventLoop;
-  this.deps = deps;
+const Periodic = require( 'Periodic' );
+const Listener = require( 'Listener' );
+
+// Encoding helper functions
+function encode( floatIn )
+{
+    return parseInt( floatIn * 1000 );
 }
-diveprofile.prototype.start = function start() {
-  var self = this;
-  this.deps.globalEventLoop.on('mcu.status', function (data) {
-    var watertype;
-    if ('dtwa' in data) {
-      watertype = {
-        raw: data.dtwa,
-        watertype: data.dtwa.toString() == '1' ? 'salt' : 'fresh'
-      };
-    }
-    if ('settings' in data) {
-      if (data.settings.water_type) {
-        watertype = {
-          raw: data.settings.water_type,
-          watertype: data.settings.water_type.toString() == '1' ? 'salt' : 'fresh'
-        };
-      }
-    }
-    if (watertype !== undefined) {
-      self.deps.cockpit.emit('plugin.diveprofile.watertype', watertype);
-      self.deps.config.preferences.set('plugin:diveprofile:water_type', watertype);
+
+function decode( intIn )
+{
+    return ( intIn * 0.001 );
+}
+
+class DiveProfile
+{
+  constructor(name, deps) 
+  {
+    console.log( 'DiveProfile plugin loaded' );
+    var self = this;
+
+    // Comm buses
+    this.globalBus  = deps.globalEventLoop;
+    this.cockpitBus = deps.cockpit;
+
+    // Plugin settings
+    this.settings = {};
+
+    // Target settings
+    this.targetWaterType  = 0;      // 0 = Fresh, 1 = Salt
+
+    // MCU's reported settings
+    this.mcuWaterType     = NaN;
+    this.zeroDepthAck     = false;
+
+    // State information
+    this.depth        = 0;    // meters
+    this.pressure     = 0;    // kPa
+    this.temperature  = 0;    // celsius
+    this.waterType    = "Freshwater";
+
+    this.SyncSettings = new Periodic( 100, "timeout", function()
```

Comment:
The concern around continue to send until a value is confirmed feels like a concern better handled lower in the framework.  Recording as a note:
Would this be better?
```Javascript
//Assumption: That the until() would only allow the latest emit of a message to be retried.

self.globalBus.emit(' mcu.SendCommand',command,target).until('settings-change.diveProfile',function(diveProfileSettings, originalMessage){return diveProfileSettings.diveProfile.waterType === target}
```